src/dataframe/dataframe.jl

```diff
@@ -544,13 +536,7 @@ for T in (:AbstractVector, :Not, :Colon)
             return df
         end
         x = df[!, col_ind]
-        try
-            x[row_inds] = v
-        catch
-            insert_multiple_entries!(df, v, axes(df, 1)[row_inds], col_ind)
-            Base.depwarn("implicit vector broadcasting in setindex! is deprecated; " *
```

Comment:
Maybe we could continue throwing an explicit error in the future? What error is printed without this?

Reply:
It is a standard error that Base prints, which I think is pretty clear in this case (as vector dimensions do not match):
```
julia> df[1:2, 1] = [1]
ERROR: DimensionMismatch("tried to assign 1 elements to 2 destinations")
```

Reply:
now if you see this error and know Base, it is obvious to write `df[1:2, 1] .= [1]` i.e.:
```
julia> x = ones(3,3)
3×3 Array{Float64,2}:
 1.0  1.0  1.0
 1.0  1.0  1.0
 1.0  1.0  1.0

julia> x[1:2, 1] = [1]
ERROR: DimensionMismatch("tried to assign 1-element array to 2×1 destination")

julia> x[1:2, 1] .= [2]
2-element view(::Array{Float64,2}, 1:2, 1) with eltype Float64:
 2.0
 2.0

julia> x
3×3 Array{Float64,2}:
 2.0  1.0  1.0
 2.0  1.0  1.0
 1.0  1.0  1.0
```
so I think it is enough to be as explicit as Base with error messages.